sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -306,7 +307,21 @@ def _read_side_inputs(self, tags_and_types):
         if not isinstance(si, operation_specs.WorkerSideInputSource):
           raise NotImplementedError('Unknown side input type: %r' % si)
         sources.append(si.source)
-      iterator_fn = sideinputs.get_iterator_fn_for_sources(sources)
+        # The tracking of time spend reading and bytes read from side inputs is
+        # behind an experiment flag to test performance impact.
+        experiments = RuntimeValueProvider(
+            'experiments', str, '').get().split(',')
+        if 'sideinput_io_metrics' in experiments:
```

Comment:
<!--new_thread; commit:ee3c5cfd0adc6bf2848f2010c0beec4d9ace5123; resolved:0-->
Can you do this outside the loop so we don't need to do the RuntimeValueProvider call each time?